cpu/atmega_common/periph/rtt.c

```diff
@@ -272,18 +423,30 @@ void rtt_clear_alarm(void)
 
 void rtt_poweron(void)
 {
+#ifdef SCCR0
+    SCCR0 |= (1 << SCEN);
+#else
     power_timer2_enable();
+#endif
 }
 
 void rtt_poweroff(void)
 {
+#ifdef SCCR0
+    SCCR0 &= ~(1 << SCEN);
+#else
     power_timer2_disable();
+#endif
 }
 
+#ifdef SCCR0
+ISR(SCNT_OVFL_vect)
+#else
 ISR(TIMER2_OVF_vect)
+#endif
 {
```

Comment:
Just make this two separate functions - the shared code is only 3 lines, one of them being `atmega_exit_isr()`.